### ISSUE TITLE
[BUG] "Ghost" incidents on line 0

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#357]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-alpha.5

### Priority

Major

### Current Behavior

On certain issues, when viewing the _Affected files_ section, there appear to be an empty incident:
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/90bbd534-05be-4b44-a360-3e775fa12354)

The second incident is correct.
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/55c782e2-f53b-43dd-a718-285d34ebcbb8)

### Expected Behavior

No empty incidents on line 0 should appear.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. In the given environment, look for the issue _Replace the Java EE version with the Jakarta equivalent_ with rule `javaee-to-jakarta-namespaces-00033`:
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/73bc435a-32e7-42e3-b86a-31ea875f5030)
3. Click on "View affected files".
4. Click on the file `/addon/source/sample/daytrader-ee7-ejb/src/main/resources/META-INF/persistence.xml` to see an example.

### Environment

```markdown
* Fedora 38
* minikube version: v1.31.2
* Konveyor 0.3.0-alpha.5
* [sample.daytrader7 app (master branch)](https://github.com/WASdev/sample.daytrader7)
* Analysis configuration:
** Source code + dependencies
** eap8, quarkus, openjdk17, azure, cloud-readiness, camel3
** Application and internal dependencies only
** No custom rules
** No additional advanced options
```

**Comment by djzager** [comment]
It turns out that neither the [xmlquery](https://github.com/antchfx/xmlquery) or [jsonquery](https://github.com/antchfx/jsonquery) dependencies we rely on to find these patterns keep track of the location in the document where the match was found. Doing this retroactively, that is after having found the respective "node", isn't practical as a simple grep for the inner text of the node may (or some similar algorithm) would not necessary give us the correct location. Additionally, it isn't safe for us to just return the entire file as a codeSnip because there are no guarantees on the maximum size of the document.

**tl;dr** the UI issue will cover the rendering portion of this --> We don't want to show empty snippets. From the analyzer perspective, this is a duplicate of #254 that would be nice for us to have but would need to be prioritized for a future release.

**Comment by djzager** [comment]
Closing this as a duplicate of #254 that we can prioritize in a future release.
